samples/ComputeSharp.SwapChain/Shaders/FourColorGradient.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace ComputeSharp.SwapChain.Shaders
+{
+    /// <summary>
+    /// A shader which interpolates four different colors into a rotating gradient pattern.
+    /// Ported from <see href="https://www.shadertoy.com/view/7lc3R4"/>.
+    /// <para>Created by Inigo Quilez.</para>
+    /// <para>License Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License.</para>
+    /// </summary>
+    [AutoConstructor]
+    internal readonly partial struct FourColorGradient : IPixelShader<float4>
+    {
+        /// <summary>
+        /// The current time since the start of the application.
+        /// </summary>
+        public readonly float time;
+
+        // Colors to blend together
+        private static readonly float3 colorOne = new(0.999f, 0.999f, 0.999f);
+        private static readonly float3 colorTwo = new(0.999f, 0f, 0f);
+        private static readonly float3 colorThree = new(0f, 0.999f, 0f);
+        private static readonly float3 colorFour = new(0f, 0f, 0.999f);
+
+        private static float2x2 Rotate(float a)
+        {
+            float c = Hlsl.Cos(a);
+            float s = Hlsl.Sin(a);
+
+            return new(c, s, -s, c);
+        }
+
+        private static float2 Hash(float2 p)
+        {
+            p = new float2(Hlsl.Dot(p, new float2(2127.1f, 81.17f)), Hlsl.Dot(p, new float2(1269.5f, 283.37f)));
+
+            return Hlsl.Frac(Hlsl.Sin(p) * 43758.5453f);
+        }
+
+        private static float Noise(float2 p)
+        {
+            float2 i = Hlsl.Floor(p);
+            float2 f = Hlsl.Frac(p);
+
+            float2 u = f * f * (3.0f - 2.0f * f);
```

Comment:
Nit: unnecessary blank line.
```suggestion
            float2 u = f * f * (3.0f - 2.0f * f);
```